src/Stackctl/Spec/Discover.hs

```diff
@@ -94,6 +98,39 @@ checkForDuplicateStackNames =
 
     exitFailure
 
+-- | Warn if a 'StackSpec' depends on a name not in the given 'StackName's
+--
+-- The 'StackName's are built from all specs, but we only run this with specs
+-- that are filtered in.
+--
+-- NB. This function is written so it can easily be made into a fatal error
+-- (like 'checkForDuplicateStackNames'), but we only warn for now.
+checkForUnknownDepends :: MonadLogger m => [StackName] -> StackSpec -> m ()
+checkForUnknownDepends known spec =
+  traverse_ reportUnknownDepends
+    $ NE.nonEmpty
+    $ filter (`notElem` known)
+    $ stackSpecDepends spec
+ where
+  reportUnknownDepends depends = do
+    for_ depends $ \depend -> do
+      let (nearest, _distance) =
+            minimumBy (comparing snd)
+              . map (id &&& getDistance depend)
+              $ known
+
+      logWarn
+        $ "Stack lists dependency that does not exist"
+        :# [ "dependency"
+              .= ( unStackName (stackSpecStackName spec)
+                    <> " -> "
+                    <> unStackName depend
+                 )
+           , "hint" .= ("Did you mean " <> unStackName nearest <> "?")
```

Comment:
optional, but it might be helpful to print *all* of the stack names like:

`did you mean 'foo'? (valid options include 'foo', 'bar', 'baz')

Reply:
That list would be 61 elements long in `infra` when called against Dev (similar for Prod). I think it makes it too big.

Reply:
jeepers yeah, that's a good point. NVM